examples/src/main/scala/examples/spv/KLS16Proof.scala

```diff
@@ -71,8 +78,8 @@ object KLS16ProofSerializer extends Serializer[KLS16Proof] {
       Bytes.concat(Shorts.toByteArray(bytes.length.toShort), bytes)
     })
     Bytes.concat(Array(obj.m.toByte, obj.k.toByte, obj.i.toByte),
-      Shorts.toByteArray(obj.suffix.head.bytes.length.toShort),
-      obj.suffix.head.bytes,
+      Shorts.toByteArray(obj.suffix.headOption.map(_.bytes.length.toShort) getOrElse 0),
+      obj.suffix.headOption.map(_.bytes) getOrElse Array[Byte](),
```

Comment:
Is this change correct?

Reply:
I think so. @kushti, could you double check?

Reply:
Actually, I'm not so sure... After all, having no head is not necessarily the same thing as having a head that is an empty array of bytes. And your code identifies these two possibly different things.

Reply:
Ok, I can take this change back and the one below too.
Besides this, I have no more comments. I will revert this change and send the final PR to review.
Thanks @ceilican :)